client/src/Week.jsx

```diff
@@ -0,0 +1,119 @@
+import React, {Component} from 'react';
+
+class Week extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    var checkInOutStyle = {
+      backgroundColor: 'black',
+      color: 'white'
+    };
+    var rangeStyle = {
+      backgroundColor: 'grey',
+      color: 'black'
+    };
+    var normalDateStyle = {
+      backgroundColor: 'white',
+      color: 'black'
+    };
+    var hoveredDateStyle = {
+      backgroundColor: 'cornflowerBlue',
+      color: 'white'
+    };
+    var blockedStyle = {
+      textDecoration: 'line-through',
+      color: 'lightGrey'
+    };
+    var checkOutOnlyStyle = {
+      color: 'grey'
+    };
+    var checkOutOnlyHoverStyle = {
+      backgroundColor: 'lightGrey',
+      color: 'grey'
+    };
+    var dateClicked = this.props.dateClicked;
+    var changedHoveredDate = this.props.changedHoveredDate;
+
+
+
+    return (<tr>
+      {this.props.thisWeek.map((item) => {
+        var chosenStyle = normalDateStyle;
+        var choosable = true;
+
+        var itemDate = new Date(item);
+        var dateIsAvailable = false;
```

Comment:
Descriptive variable naming, that is really cool!
There is a naming convention style for Boolean variables which might be worth exploring.
'dateIsAvailable' vs 'isDateAvailable'

A helpful discussion on the same! :)
[Boolean Variable](https://stackoverflow.com/questions/1227998/naming-conventions-what-to-name-a-boolean-variable)